docs/site/Extending-OpenAPI-specification.md

```diff
@@ -0,0 +1,154 @@
+  modifySpec(spec: OpenApiSpec): OpenApiSpec {
+    const InfoPatchSpec = {
+      info: {title: 'LoopBack Test Application', version: '1.0.1'},
+    };
+    // the example calls a default helper function to merge the fragment spec.
+    const mergedSpec = defaultMergeFn(spec, InfoPatchSpec);
+    return mergedSpec;
+  }
+}
+```
+
+- The class is decorated with a binding template `asSpecEnhancer`.
+- The enhancer has a name as `info`. Name can be used to retrieve a certain
+  enhancer (explained in the
+  [extension point section](#oas-enhancer-service-as-extension-point)).
+- The enhancer changes the current specification's `info` object in function
+  `modifySpec`.
+- It calls [`defaultMergeFn`](#default-merge-function) to merge the
+  specification fragment into the current spec.
+
+### Default Merge Function
+
+Since `modifySpec` has full access to the current spec, it can perform any
+operation: merge, delete, or more complicated changes. This is totally
+determined by the extension contributor.
+
+To apply the basic merging, we provide a default helper function called
+`defaultMergeFn` that leverages
+[`json-merge-patch`](https://github.com/pierreinglebert/json-merge-patch) to
+merge two json objects. You can find its usage in the
+[previous section](#adding-a-new-oas-enhancer)
+
+### Registering an Enhancer
+
+After decorating your enhancer properly with `@bind(asSpecEnhancer)`, you can
+bind it to your application as follow:
+
+```ts
+import {createBindingFromClass} from '@loopback/core';
+import {InfoSpecEnhancer} from './enhancers/infoSpecEnhancer';
+
+class MyApplication extends RestApplication {
+  constructor() {
+    super();
+    this.add(createBindingFromClass(InfoSpecEnhancer));
+  }
+}
+```
+
+## OAS Enhancer Service as Extension Point
+
+The OAS enhancer extension point is created in package `@loopback/openapi-v3`.
+It organizes the registered OAS enhancers, and as the first implementation, it
+provides APIs to either apply one enhancer by name, or apply all enhancers
+automatically.
+
+### Registering an Enhancer Service
+
+You can bind the OAS enhancer extension point to your app via key
+`OAS_ENHANCER_SERVICE`:
+
+```ts
+import {RestApplication} from '@loopback/rest';
+import {OASEnhancerService, OAS_ENHANCER_SERVICE} from '@loopback/openapi-v3';
+
+class MyApplication extends RestApplication {
+  constructor() {
+    super();
+    this.add(
+      createBindingFromClass(OASEnhancerService, {
+        key: OAS_ENHANCER_SERVICE,
+      }),
+    );
+  }
+
+  // define a function to return a spec service by the same key
+  getSpecService() {
+    return this.get(OAS_ENHANCER_SERVICE);
+  }
+}
+```
+
+### Apply Registered Enhancers
+
+To automatically apply all the registered enhancers, call `applyAllEnhancers`:
+
+```ts
+await app.getSpecService.applyAllEnhancers();
+```
+
+_In the future we will support apply enhancers by a custom sequence. The
+sequence will be determined by a combination of group names and the alphabet
```

Comment:
nitpick: `and the alphabet order` -> `and alphabetical order`
Can you clarify what you mean by `a combination of group names`?

Reply:
@nabdelgadir See our discussion in https://github.com/strongloop/loopback-next/pull/4258#discussion_r355542472, same approach as the applying the observers :)